ui/core/components/character_stats.tsx

```diff
@@ -58,8 +59,23 @@ export class CharacterStats extends Component {
 			this.valueElems.push(valueElem);
 		});
 
+		if(this.shouldShowMeleeCritCap(player)) {
+			const row = (
+			<tr
+				className='character-stats-table-row'
+			>
+				<td className="character-stats-table-label">Melee Crit Cap</td>
+				<td className="character-stats-table-value"></td>
+			</tr>);
+
+			table.appendChild(row);
+			this.meleeCritCapValueElem = row.getElementsByClassName('character-stats-table-value')[0] as HTMLTableCellElement;
+		} else {
+			this.meleeCritCapValueElem = undefined;
+		}
+
 		this.updateStats(player);
-		TypedEvent.onAny([player.currentStatsEmitter, player.sim.changeEmitter]).on(() => {
+		TypedEvent.onAny([player.currentStatsEmitter, player.sim.changeEmitter, player.talentsChangeEmitter]).on(() => {
```

Comment:
Character stats now update on talent change to compute Elemental Devastation's effect on crit cap